sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImpl.java

```diff
@@ -45,6 +43,8 @@
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.annotations.VisibleForTesting;
```

Comment:
Why only these two use the vendored version? Any hints on the 'ideal practice' in this case @kennknowles?
I suppose we should stick to the vendored imports for Beam's classes in all cases, otherwise we will have a harder time to guarantee correctness, no?

@kewne sorry if this is taking longer than intended, vendoring and using vendored deps is a new Beam area so still some rough areas, but hopefully this will make it ASAP or at least as part of 2.10.x

Reply:
The idea is that if Guava is used just for helpful utilities, we use the vendored version. So in a perfect world, we would always use just the vendored version. The only change is that Cassandra uses Guava classes on its API, so we cannot use the vendored version there.

Reply:
Can't we just use In all the Beam related code the vendored dependency and then let the Cassandra java driver code run with its own guava version?
I am assuming of course that the dependencies (e.g. `sdks/java/core`) are already using the vendored version of guava. Is this the idea in the future or I am misunderstanding something.

Reply:
Yes, that is what I was trying to say. But I think the Beam CassandraIO code needs to _also_ use the Cassandra-related Guava version for a couple things.

Reply:
Well in that case CassandraIO will leak its version of Java which is bad but not as bad as not working. @kewne can you do these changes (change the imports manually to use the vendored version of guava in the IO files `org.apache.beam.vendor.guava.v20_0.com.google.common.`) and confirm if the tests pass now.

Reply:
@kennknowles do we plan to publish also a vendored guava artifact to solve the issues mentioned above? (talking about the guava 5 times in direct runner).

Reply:
Yes, it is here: https://mvnrepository.com/artifact/org.apache.beam/beam-vendor-guava-20_0. The idea is that we port all shaded Guava in all modules to this and then stop shading. Is that what you mean?

Reply:
Oh didn't know this was already published, I suppose this was part of the first vote. We need to document and maybe make more clear the vendored artifacts vote/release validation process,
The idea is that now in all modules e.g. direct-runner include instead of all the `com.google.common.*` imports the `org.apache.beam.vendor.guava.v20_0.com.google.common.*` ones? If so we will need to do this manually or this is somehow automatized when packaging with gradle and its magic? Tell me if you need help with that I think this is important enough to work on it (at least reduce the 5 times guava in direct runner present).

Reply:
I agree we need to work on the release process for the vendored bits. You proposal is exactly right. Once we manually port e.g. direct-runner to use `org.apache.beam.vendor.guava.v20_0.com.google.common` then we turn off shading of Guava for the direct-runner. For most modules, once we do this we can turn of shading completely, because they only shade Guava. It will speed & simplify the build a lot and work better with IntelliJ.

Reply:
@kennknowles Mmmm I will like to tackle that soon, however I am not sure if we are ready to start this now or better try to move things into a more recent version of guava because of [BEAM-6267](https://jira.apache.org/jira/browse/BEAM-6267). including the version in the imports will guarantees us some noisy commits every time we upgrade versions. WDYT? Or do we care about the security issue afterwards?

Reply:
I think the global search/replace is easy either way. I don't think the security issue matters for us, but we should just update anyhow.

Reply:
Yes the security issue has zero impact in our current codebase, but people run these automated CVE checking tools that detect libraries and use this as a base to accept or not to use products so we should take care at some moment.
I did the replacement that I suggested and found that it still breaks because some of the dependencies are still leaking guava 20.0 (it is not well shaded somewhere), This needs further investigation.